src/Engine/Manager/PhysicsManager.cpp

```diff
@@ -87,20 +94,34 @@ void PhysicsManager::Update(float deltaTime) {
     }
 }
 
+void PhysicsManager::UpdateEntityTransforms() {
+    std::vector<Component::Physics*> physicsObjects = physicsComponents.GetAll();
+    for (Component::Physics* physicsComp : physicsObjects) {
+        if (physicsComp->IsKilled() || !physicsComp->entity->enabled)
+            continue;
+
+        Entity* entity = physicsComp->entity;
+
+        auto trans = physicsComp->GetRigidBody().GetRigidBody()->getWorldTransform();
+        entity->position = Physics::btToGlm(trans.getOrigin());
+        entity->rotation = glm::eulerAngles(Physics::btToGlm(trans.getRotation()));
+    }
+}
+
+void PhysicsManager::OnTriggerEnter(Component::Physics* triggerBody, Component::Physics* object, std::function<void()> callback) {
+    auto trigger = MakeTrigger(triggerBody);
+    trigger->OnEnter(object, callback);
+}
+
 Physics::RigidBody* PhysicsManager::MakeRigidBody(Physics::Shape* shape, float mass) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The function 'MakeRigidBody' is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)